bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBridgeHandler.java

```diff
@@ -0,0 +1,1131 @@
+        List<String> values = new ArrayList<>();
+        try {
+            InputSource inputXML = new InputSource(new StringReader(xmlResponse));
+            XPath xPath = XPathFactory.newInstance().newXPath();
+            NodeList nodes = (NodeList) xPath.evaluate(xpathExp, inputXML, XPathConstants.NODESET);
+
+            for (int i = 0; i < nodes.getLength(); i++) {
+                values.add(nodes.item(i).getNodeValue());
+            }
+
+        } catch (XPathExpressionException e) {
+            logger.error("XPathExpression exception:", e);
+        }
+        return values;
+    }
+
+    private Request sendRequestBuilder(String url, HttpMethod method) {
+        return this.httpClient.newRequest(url).agent("NextGenForIPhone/16565 CFNetwork/887 Darwin/17.0.0")
+                .method(method).header(HttpHeader.ACCEPT_LANGUAGE, "en-us").header(HttpHeader.ACCEPT, "*/*")
+                .header(HttpHeader.ACCEPT_ENCODING, "gzip, deflate").version(HttpVersion.HTTP_1_1)
+                .header(HttpHeader.CONNECTION, "keep-alive").header(HttpHeader.HOST, "www.haywardomnilogic.com:80")
+                .timeout(10, TimeUnit.SECONDS);
+    }
+
+    private synchronized String httpXmlResponse(String urlParameters) throws Exception {
+        int status;
+        String statusMessage;
+        String urlParameterslength = Integer.toString(urlParameters.length());
+
+        ContentResponse httpResponse = sendRequestBuilder(config.hostname, HttpMethod.POST)
+                .content(new StringContentProvider(urlParameters), "text/xml; charset=utf-8")
+                .header(HttpHeader.CONTENT_LENGTH, urlParameterslength).send();
+
+        status = httpResponse.getStatus();
+        String xmlResponse = httpResponse.getContentAsString();
+
+        if (!(evaluateXPath("/Response/Parameters//Parameter[@name='StatusMessage']/text()", xmlResponse).isEmpty())) {
+            statusMessage = evaluateXPath("/Response/Parameters//Parameter[@name='StatusMessage']/text()", xmlResponse)
+                    .get(0);
+
+        } else {
+            statusMessage = httpResponse.getReason();
+        }
+
+        if (status == 200) {
+            if (logger.isTraceEnabled()) {
+                logger.trace("{} Hayward http command: {}", getCallingMethod(), urlParameters);
+                logger.trace("{} Hayward http response: {} {}", getCallingMethod(), statusMessage, xmlResponse);
+            } else if (logger.isDebugEnabled()) {
+                logger.debug("{} Hayward http response: {}", getCallingMethod(), statusMessage);
+            }
+            return xmlResponse;
+        } else {
+            if (logger.isErrorEnabled()) {
+                logger.error("{} Hayward http command: {}", getCallingMethod(), urlParameters);
+                logger.error("{} Hayward http response: {}", getCallingMethod(), status);
+            }
+            return "";
```

Comment:
Rather than updating the thing status upon a single failure, the initPolling method waits for 5 consecutive failures, then calls the initialize method.  If the subsequent login attempt fails, the thing is set offline.  My intent here is to build in a little network tolerance before the binding goes offline.